grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -34,18 +34,23 @@ class UserService {
 
     def updateUser(String userId, String currentUserId, Map params) {
 
-        def userInstance = User.get(userId)
+        User userInstance = User.get(userId)
+        List<Role> updatedRoles = params.roles ? Role.findAllByIdInList(params.list("roles")) : []
```

Comment:
It actually doesn't matter, because when binding it has to fetch it anyway (but maybe only a proxy (?)), but you can check if just:
```groovy
userInstance.roles = params.list("roles")
```
would be enough

Reply:
this looks fine to me.
At first I thought that we might want to use `bindData(userInstance, params)` that way it will properly deal with binding and fetching all of the appropriate data for us, but apparently we can not use bindData outside of the controller _(at least it looks that way after tested it localy)_ so I am wondering if we maybe should move the binding part to controller and then pass the userInstance as a parameter and not a `user.id`
```groovy
bindData(userInstance, params)
 userInstance = userService.updateUser(userInstance, session.user.id, params)

```

Reply:
I also thought about `bindData`, but you're right that we can't use it outside the controller. For me, it looks weird that we are passing a user instance as an argument and then we are assigning the result to the user instance